core/src/main/scala/za/co/absa/spline/harvester/LineageHarvester.scala

```diff
@@ -142,37 +142,45 @@ class LineageHarvester(
     })
   }
 
+  private trait StackItem
+  private case class NewItem(plan: LogicalPlan, parentBuilder: OperationNodeBuilder) extends StackItem
+  private case class ProcessedItem(builder: OperationNodeBuilder) extends StackItem
+
   private def createOperationBuildersRecursively(rootOp: LogicalPlan): Seq[OperationNodeBuilder] = {
```

Comment:
Does this method return builders in a topological order? If so, what piece of code is responsible for the node precedence?